### ISSUE TITLE
Please find me which part I did wrong

### ISSUE THREAD

**Issue by miridih-jyjang** [THUDM/WebRL#56]
# 1. SFT Training
I used llama-3.1-8b (pretraining) as a initial weight
--- 
cd LLaMA-Factory
bash run.sh examples/train_full/llama3_full_policy_web.yaml
---
<img width="2048" height="1022" alt="Image" src="https://github.com/user-attachments/assets/0ed2ade5-7889-4b57-b646-cf5458bf5bbb" />
## Training data trajectory
Total trajectory : 1124
Total unique task : 1113
 
# 2. Scoring
After SFT Training, I got following scores
## 2.a Run vLLM
---
MODEL_PATH="/data/shared/jyjang/WebRL/llama3-8b-sft/"
HOST=<your_ip_address>
PORT=<your_port_number>
GPU_MEMORY_UTILIZATION="0.9"
MAX_MODEL_LEN="16384"

export CUDA_VISIBLE_DEVICES=0 # Use available GPUs
export VLLM_USE_MODELSCOPE="False"

echo "Starting VLLM server..."
echo "Model: $MODEL_PATH"
echo "Host: $HOST"
echo "Port: $PORT"

python -m vllm.entrypoints.openai.api_server \
    --model $MODEL_PATH \
    --host $HOST \
    --port $PORT \
    --gpu-memory-utilization $GPU_MEMORY_UTILIZATION \
    --max-model-len $MAX_MODEL_LEN \
    --served-model-name "webrl-llama3-8b" \
    --trust-remote-code \
    --disable-log-requests \
    --tensor-parallel-size 1 \
    --dtype auto

---
## 2.b Run Environment
Run each Docker based websites

## 2.c Rollout trajectories with Test data
---
/workspace/VisualAgentBench/VAB-WebArena-Lite# ./wa_parallel_run_webrl.sh 
---
## 2.d Scoring with test data
src file:  result/llama-3.1-8b-sft/test_data/
successed:  44 /  164 (812) 
**partial accuracy:   26.83** -> Too high compared to the paper (20.6)
overall accuracy:   26.67
--------

('shopping_admin',) 22.9
('map',) 19.2
('shopping',) 15.6
('reddit',) 57.9
('gitlab',) 40.0
('map', 'wikipedia') 0.0
('wikipedia', 'map') 33.3
('gitlab', 'reddit') 0.0
('shopping', 'reddit') 0.0
('reddit', 'gitlab') 0.0
('map', 'shopping_admin') 0.0

('shopping_admin',) [95, 157, 374, 470, 497, 538, 678, 710] 8
('map',) [37, 75, 88, 139, 367] 5
('shopping',) [144, 276, 324, 354, 466, 514, 521] 7
('reddit',) [27, 404, 582, 599, 601, 612, 641, 645, 652, 714, 733] 11
('gitlab',) [44, 156, 258, 311, 357, 415, 476, 485, 534, 745, 748, 811] 12
('map', 'wikipedia') [] 0
('wikipedia', 'map') [741] 1
('gitlab', 'reddit') [] 0
('shopping', 'reddit') [] 0
('reddit', 'gitlab') [] 0
('map', 'shopping_admin') [] 0

## 2.e Rollout with Training data 
src file:  result/llama-3.1-8b-sft/train_data
successed: 1115 / 1115 (812)
**partial accuracy:   100.0**
overall accuracy:  675.76
--------

('shopping_admin',) 100.0
('map',) 100.0
('shopping',) 100.0
('reddit',) 100.0
('gitlab',) 96.7
('map', 'wikipedia') 100.0
('wikipedia', 'map') 100.0
('gitlab', 'reddit') 100.0
('shopping', 'reddit') 100.0
('reddit', 'gitlab') 100.0
('map', 'shopping_admin') 100.0

('shopping_admin',) [4, 15, 43, 65, 77, 95, 109, 115, 123, 127, 131, 157, 196, 202, 211, 215, 247, 288, 348, 374, 423, 454, 458, 464, 470, 488, 491, 497, 505, 538, 548, 678, 704, 710, 768] 35
('map',) [7, 20, 33, 37, 56, 58, 71, 75, 82, 88, 93, 98, 139, 155, 220, 221, 236, 250, 254, 287, 367, 369, 381, 382, 383, 762] 26
('shopping',) [23, 48, 96, 117, 118, 125, 144, 149, 162, 167, 190, 225, 227, 235, 240, 270, 276, 283, 285, 300, 313, 321, 324, 333, 335, 354, 361, 368, 376, 384, 386, 387, 431, 440, 466, 506, 509, 514, 516, 521, 528, 574, 657, 693, 798] 45
('reddit',) [27, 69, 401, 404, 582, 599, 601, 605, 612, 619, 626, 631, 641, 645, 652, 714, 720, 729, 733] 19
('gitlab',) [44, 103, 156, 169, 173, 182, 205, 258, 259, 296, 311, 318, 349, 357, 392, 415, 419, 448, 476, 485, 524, 534, 567, 577, 668, 745, 748, 809, 811] 29
('map', 'wikipedia') [97] 1
('wikipedia', 'map') [268, 426, 741] 3
('gitlab', 'reddit') [566, 791] 2
('shopping', 'reddit') [673] 1
('reddit', 'gitlab') [682, 686] 2
('map', 'shopping_admin') [760] 1
---
My question is

1. The sft test score is relatively too high compared to the paper. is there anything i did wrong? or is it reasonable to get that 26.83 score in stage 0?
2. I get zero failure trajectories for the F buffer, which I couldn't move forward. Is there anything i did wrong?

Thanks for reading a long message.

**Comment by QZH-777** [comment]
I didn't find anything wrong with your process.

For your first question, you can start by testing the scores of llama3.1-instruct to see if they match those reported in the paper, which would help rule out any environmental factors. Also, based on the scores, your current results on GitLab are significantly higher than those in the paper. My guess is that there may be some randomness, so it might be helpful to run a few more experiments. That said, the score of 26.8 is acceptable and can still be further improved by applying the RL method afterwards.

For your second question, I'd like to confirm how you’re determining correctness—is it based on our reward model? Additionally, you could try setting the temperature to 1 and sampling several more times, which may help you identify tasks that the model isn't handling well. Those tasks can be used for the next phase.

**Comment by miridih-jyjang** [comment]
@QZH-777 Thanks for the advice.
For question 1. I found that I used Llama-3.1-8B instead of Llama-3.1-8B-instruct so that the score would get 26.8. When I used Llama-3.1-8B-instruct, the Behavior Cloning(BC, or SFT) result decreases back to 14.55, which is a lot lower than the paper.

For question 2. even though I changed the temperature from 0.0 -> 1.0 the rollout result for training data still get 100% accuracy, which I still couldn't get the Failure cases. I cannot point out which point I am misunderstanding.
For reward, I used webarena reward, which is line 3 in the paper algorithm 1.

Please give me any advice if possible, Thanks

<img width="687" height="531" alt="Image" src="https://github.com/user-attachments/assets/d846e7a3-2c19-495a-8b2e-51e23ee9973e" />

**Comment by miridih-jyjang** [comment]
I found out that I misinterpreted WebArena-Lite reward function. Instead of WebArena-Lite reward function, I noticed in https://github.com/THUDM/WebRL/issues/40 that I should use ORM instead of WebArena-Lite reward Function. 
I Also recommend to fix the line 3 if you don't have plans to open the WebArena-Lite reward function.
So it could be closed!
